testsuite/mpitests/issue-600.sli

```diff
@@ -40,13 +40,8 @@
 
 % The following test needs the model hh_psc_alpha_gap, so
 % this test should only run if we have GSL
-is_threaded not 
-statusdict/have_gsl :: not
-or 
-{ 
-  exit_test_gracefully
-} 
-if
+is_threaded not {(no_openmp) exit_test_gracefully_reason} if
+statusdict/have_gsl :: not {(no_gsl) exit_test_gracefully_reason} if
```

Comment:
We have a small number of skipping conditions which lead to a lot of repeated code. I think we could make things much tidier by introducing functions for each of the cases, e.g.,
```
/skip_if_not_threaded { is_threaded not { /no_openmp exit_test_gracefully } if } def
```
Then, in this case, the test code would just read
```
skip_if_not_threaded
skip_if_without_gsl
```